src/main/java/org/mule/extension/file/common/api/command/ReadCommand.java

```diff
@@ -48,6 +48,23 @@
     return read(config, filePath, lock);
   }
 
+  /**
+   * Reads files under the considerations of {@link FileSystem#read(FileConnectorConfig, String, boolean)}
+   *
+   * @param config the config that is parameterizing this operation
+   * @param filePath the path of the file you want to read
+   * @param lock whether or not to lock the file.
+   * @param timeBetweenSizeCheck wait time between size checks to determine if a file is ready to be read in milliseconds.
+   * @param lockTimeout time in nanoseconds that the operation will spend trying to lock the file.
+   * @return An {@link Result} with an {@link InputStream} with the file's content as payload and a {@link FileAttributes} object
+   *         as {@link Message#getAttributes()}
+   * @throws IllegalArgumentException if the file at the given path doesn't exist
+   */
+  default Result<InputStream, A> read(FileConnectorConfig config, String filePath, boolean lock, Long timeBetweenSizeCheck,
+                                      long lockTimeout) {
+    return read(config, filePath, lock);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "read"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule-file-commons&pullRequest=56&issues=AWm2Sxs0otIXnmsUPRcF&open=AWm2Sxs0otIXnmsUPRcF)

[ISSUE_KEY]: AWm2Sxs0otIXnmsUPRcF